tests/components/ecobee/test_humidifier.py

```diff
@@ -0,0 +1,64 @@
+"""The test for the Ecobee thermostat module."""
+from unittest import mock
+
+import pytest
+
+from homeassistant.components.ecobee import humidifier as ecobee
+
+
+@pytest.fixture
+def ecobee_fixture():
+    """Set up ecobee mock."""
+    vals = {
+        "name": "ecobee",
+        "runtime": {
+            "desiredHumidity": 30,
+        },
+        "settings": {
+            "hasHumidifier": True,
+            "humidifierMode": "auto",
+        },
+    }
+    mock_ecobee = mock.Mock()
+    mock_ecobee.__getitem__ = mock.Mock(side_effect=vals.__getitem__)
+    mock_ecobee.__setitem__ = mock.Mock(side_effect=vals.__setitem__)
+    return mock_ecobee
+
+
+@pytest.fixture(name="data")
+def data_fixture(ecobee_fixture):
+    """Set up data mock."""
+    data = mock.Mock()
+    data.ecobee.get_thermostat.return_value = ecobee_fixture
+    return data
+
+
+@pytest.fixture(name="thermostat")
+def humidifier_fixture(data):
+    """Set up ecobee thermostat object."""
+    return ecobee.EcobeeHumidifier(data, 1)
```

Comment:
Please start all tests with the core interfaces and don't interact with integration details.

https://developers.home-assistant.io/docs/development_testing#writing-tests-for-integrations

Set up the integration and let it load the humidifier platform and let it add an entity. Then assert the state of the entity via the core state machine.

Reply:
Test the services via the service registry.